### PR TITLE
Added unique key for issue credential id and attestation id

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/set-up-to-5-credentials-as-favourite-9152370013517406)

### CODE REVIEW

File: src/prisma/schema.prisma [hunk 1]
```diff
@@ -2086,3 +2087,15 @@ model CharmTransaction {
   toWallet   CharmWallet? @relation("toWallet", fields: [to], references: [id], onDelete: SetNull)
   metadata   Json?        @default("{}")
 }
+
+model FavoriteCredential {
```
Comment (motechFR) [inline src/prisma/schema.prisma:2091]: Let's add a user relation here

File: src/prisma/schema.prisma [hunk 2]
```diff
@@ -2086,3 +2087,15 @@ model CharmTransaction {
...
+  id                 String            @id @default(uuid()) @db.Uuid
+  index              Int               @default(-1)
+  attestationId      String?
```
Comment (motechFR) [inline src/prisma/schema.prisma:2094]: Suggestion; let's add some comments to differentiate

// On chain
attestationId
chainId

// CharmVerse credential
issuedCredentialId
issuedCredential

// Gitcoin
gitcoinWalletAddress

File: src/prisma/schema.prisma [hunk 3]
```diff
@@ -2086,3 +2088,17 @@ model CharmTransaction {
...
+  index                Int               @default(-1)
+  chainId              Int?
+  userId               String            @db.Uuid
+  user                 User              @relation(fields: [userId], references: [id], onDelete: Cascade)
+  attestationId        String?
+  issuedCredentialId   String?           @db.Uuid
+  issuedCredential     IssuedCredential? @relation(fields: [issuedCredentialId], references: [id], onDelete: SetNull)
+  gitcoinWalletAddress String?
+
+  @@unique([issuedCredentialId, attestationId, gitcoinWalletAddress])
```
Comment (motechFR) [inline src/prisma/schema.prisma:2103]: @@unique[] defines a composite index with all 3 values

Instead here, you should @unique each individual field

File: src/prisma/schema.prisma [hunk 4]
```diff
@@ -2086,3 +2088,17 @@ model CharmTransaction {
...
+model FavoriteCredential {
+  id                   String            @id @default(uuid()) @db.Uuid
+  index                Int               @default(-1)
+  chainId              Int?
+  userId               String            @db.Uuid
+  user                 User              @relation(fields: [userId], references: [id], onDelete: Cascade)
+  attestationId        String?
+  issuedCredentialId   String?           @db.Uuid
+  issuedCredential     IssuedCredential? @relation(fields: [issuedCredentialId], references: [id], onDelete: SetNull)
+  gitcoinWalletAddress String?
```
Comment (motechFR) [inline src/prisma/schema.prisma:2101]: Please double check what format we send to the Gitcoin API (lowercase or checksummed) and make sure it's compatible with what's stored here

If lowercase works on the API, let's go with that here to align with rest of the system

Comment (Devorein) [reply src/prisma/schema.prisma:2101]: Yep it does work, I will store the lowercased version on db

**Pull request merged**